### PR TITLE
New AWS module for managing IAM groups fork from @naslanidis

### PR DESCRIPTION
**Pull request by mpostument**
##### SUMMARY
The module was forked from https://github.com/ansible/ansible/pull/21236. Because author left this module unfinished 

##### ISSUE TYPE
New Module Pull Request

##### COMPONENT NAME
iam_group

##### ANSIBLE VERSION
```
ansible 2.3.0.0
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
Module to allow managing AWS IAM groups.

As per the below PR, it was decided that the iam module should be split in to separate ones i.e. iam_role, iam_user, iam_group. This module completes the iam_group module.

I debated whether to include the user output, however as this is returned by the get_group API action and is useful I decided to keep it.

https://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.get_group
```
TASK [debug] *************************************************
ok: [localhost] => {
    "msg": {
        "changed": true, 
        "iam_group": {
            "group": {
                "arn": "arn:aws:iam::1234567890:group/test_group", 
                "create_date": "2017-02-10T09:43:01+00:00", 
                "group_id": "AGPAIDBWE123SFINE55TM", 
                "group_name": "test_group", 
                "path": "/"
            }, 
            "users": [
                {
                    "arn": "arn:aws:iam::1234567890:user/test_user1", 
                    "create_date": "2017-02-10T07:12:32+00:00", 
                    "path": "/", 
                    "user_id": "AIDAIZTPYHR4123S22YU2", 
                    "user_name": "test_user1"
                }
            ]
        }
    }
}

```

### CODE REVIEW

**Comment by mpostument** [comment]
@willthames can you review this?

**Comment by willthames** [comment]
I'd rather commit authorship was preserved when forking. This basically means taking the original work, rebasing it on latest, perhaps squashing the original commits and then continuing from there. It's probably not too hard to take what you've done and rebase it it on a rebase of @naslanidis's commits.

File: lib/ansible/modules/cloud/amazon/iam_group.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+version_added: "2.4"
+author: Nick Aslanidis, @naslanidis, Maksym Postument, @infectsoldier
+options:
+  name:
+    description:
+      - The name of the group to create.
+    required: true
+  managed_policy:
+    description:
+      - A list of managed policy ARNs (can't use friendly names due to AWS API limitation) to attach to the group.
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/iam_group.py:36]: That is a limitation but there is a workaround (see #22630)

File: lib/ansible/modules/cloud/amazon/iam_group.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(required=True, type='str'),
+            managed_policy=dict(default=[], type='list'),
+            users=dict(default=[], type='list'),
+            state=dict(default=None, choices=['present', 'absent'], required=True),
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/iam_group.py:372]: default=None is not required

**Comment by mpostument** [comment]
Probably I broke the branch. I will recreate it later

**Pull request closed without merging**